vault/core.go

```diff
@@ -916,6 +930,101 @@ func (c *Core) unsealPart(config *SealConfig, key []byte) ([]byte, error) {
 	return masterKey, nil
 }
 
+// Checks local version against data version. If data version doesn't
+// exist, it will attempt to write it. This must be called with the
+// state write lock held.
+func (c *Core) versionCheck() error {
+	// Get data version
+	rawDataVer, err := c.barrier.Get(coreDataVersionPath)
+	if err != nil {
+		c.logger.Error("core: data version check failed", "error", err)
+		c.barrier.Seal()
+		c.logger.Warn("core: vault is sealed")
+		return err
+	}
+	// If data version is empty, skip version comparison
+	if rawDataVer == nil {
+		// Handle case for secondary
+		if c.replicationState == consts.ReplicationSecondary {
+			c.logger.Warn("core: data version not found, ignoring version check")
```

Comment:
I think that this warning should actually be in the outer `if` block -- it's relevant either way.